Content.Server/Temperature/Systems/TemperatureDamageSystem.cs

```diff
@@ -0,0 +1,143 @@
+using Content.Server.Administration.Logs;
+using Content.Server.Temperature.Components;
+using Content.Shared.Alert;
+using Content.Shared.Damage;
+using Content.Shared.Database;
+using Content.Shared.Temperature.Components;
+using Content.Shared.Temperature.Systems;
+
+namespace Content.Server.Temperature.Systems;
+
+public sealed partial class TemperatureDamageSystem : EntitySystem
+{
+    [Dependency] private readonly IAdminLogManager _adminLogger = default!;
+    [Dependency] private readonly AlertsSystem _alertsSystem = default!;
+    [Dependency] private readonly DamageableSystem _damageableSystem = default!;
+    private EntityQuery<TransformComponent> _xformQuery = default!;
+    private EntityQuery<DamageableComponent> _damageQuery = default!;
+    private EntityQuery<TemperatureComponent> _temperatureQuery = default!;
+    private EntityQuery<TemperatureDamageThresholdsComponent> _thresholdsQuery = default!;
+    private EntityQuery<ContainerTemperatureDamageThresholdsComponent> _containerThresholdsQuery = default!;
+
+    /// <summary>
+    /// All of the entities that will be processed during the next temperature damage tick.
+    /// Exists because things can change an entities temperature multiple times per tick and we want to avoid double dipping on damage.
+    /// </summary>
+    private readonly HashSet<Entity<TemperatureDamageThresholdsComponent, TemperatureComponent>> _shouldUpdateDamage = new();
+
+    /// <summary>
+    /// The amount of time that has passed since the last temperature damage tick.
+    /// </summary>
+    private float _accumulatedFrameTime = 0f;
```

Comment:
This won't work, store a `TimeSpan` datafield on the comp instead and use the offset serializer.

Reply:
Done in #30515